src/main/java/org/javarosa/xpath/analysis/ReferencesMainInstanceAnalyzer.java

```diff
@@ -11,21 +11,19 @@
  */
 public class ReferencesMainInstanceAnalyzer extends XPathBooleanAnalyzer {
 
-    private String mainInstanceRoot;
-
-    public ReferencesMainInstanceAnalyzer(String instanceName, EvaluationContext ec) {
-        this(instanceName);
+    public ReferencesMainInstanceAnalyzer(EvaluationContext ec) {
+        this();
         setContext(ec);
     }
 
-    public ReferencesMainInstanceAnalyzer(String instanceName) {
+    public ReferencesMainInstanceAnalyzer() {
         super();
-        this.mainInstanceRoot = instanceName;
     }
 
     @Override
     public void doNormalTreeRefAnalysis(TreeReference treeRef) throws AnalysisInvalidException {
-        if (treeRef.getName(0).equals(mainInstanceRoot)) {
+        if (treeRef.getContextType() == TreeReference.CONTEXT_ABSOLUTE &&
```

Comment:
Small style point that this being in a method called 

"doesTreeReferenceReferToMainInstance()"

or something would be a good point of self-documentation for what this check means

Reply:
I think the name of the class basically achieves that already, but fine with adding this as well